src/System.Net.Security/tests/FunctionalTests/ClientAsyncAuthenticateTest.cs

```diff
@@ -98,7 +100,23 @@ public void ClientAsyncAuthenticate_MismatchProtocols_Fails(SslProtocols server,
             Assert.Throws(expected, () => ClientAsyncSslHelper(server, client));
         }
 
+        [Theory]
+        [MemberData("ProtocolMismatchDataSsl2Specific")]
+        [PlatformSpecific(PlatformID.Windows)]
```

Comment:
added the test case to verify the exception for sslv2. 
I don't find a way we can explicitly convert this particular exception to map to the windows.
And I think for any unsupported  protocol exception windows code would also return the authentication exception!. Here windows is returning IOException for the mismatch scenario.